From 4b238c10f4ee7fd1e6a2fab9433814d888d3d0cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Tue, 29 Oct 2019 16:48:00 +0000
Subject: [PATCH 46/50] window: Warn if O-R window workspace state is used

Override-redirect windows have no workspace by default, and can't be parent
of a top-level window, so we must check that the parent window is not an
O-R one when setting the workspace state.

https://gitlab.gnome.org/GNOME/mutter/merge_requests/895


(cherry picked from commit 2644e54c514377009a9d4d747874a55b387c6beb)
---
 src/core/window.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/window.c b/src/core/window.c
index d93e723a5..f38f0fee2 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -1306,6 +1306,7 @@ _meta_window_shared_new (MetaDisplay         *display,
                       "Putting window %s on same workspace as parent %s\n",
                       window->desc, window->transient_for->desc);
 
+          g_warn_if_fail (!window->transient_for->override_redirect);
           set_workspace_state (window,
                                should_be_on_all_workspaces (window->transient_for),
                                window->transient_for->workspace);
-- 
2.23.0

